app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  scope :not_invited_this_event, ->(ids) { where("User_id in ARRAY[?]::integer[]", ids) }
+
+  before_save { self.email = email.downcase }
+  validates :name,  presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  validates :password, presence: true, length: { minimum: 2 }
+
+
+  has_many :attendances
+  has_many :events, through: :attendances
+  has_many :events
```

Comment:
Here you are using `:events` relationship twice and that might be confusing. The user is playing two roles, one as a `creator` of the event and one as an attendee so try to use different names to them. The through relationship might be named as `attended_events`, for example.